### PR TITLE
added cluster metadata

### PR DESCRIPTION
**Pull request by fatz**
added /metadata endpoint

### CODE REVIEW

File: dcos/docs/MetadataApi.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# \MetadataApi
```
Comment (ArmandGrillet) [inline dcos/docs/MetadataApi.md]: `\` can be removed.

File: dcos/model_metadata.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * API version: 1.0.0
+ */
+
+// Code generated by OpenAPI Generator (https://openapi-generator.tech); DO NOT EDIT.
+
+package dcos
+
+type Metadata struct {
+	PUBLICIPV4 string `json:"PUBLIC_IPV4,omitempty"`
```
Comment (ArmandGrillet) [inline dcos/model_metadata.go]: Could we get `PUBLIC_IPV4` instead?

Comment (fatz) [reply dcos/model_metadata.go]: not sure how to enforce. Although better would be something like `PublicIPV4` but not sure how to influence the struct from OpenAPI description. Any suggestions?

Comment (bamarni) [reply dcos/model_metadata.go]: In the spec we have `x-go-name` for one property, but it seems to be coming from the go-swagger project, I don't see anything similar in `openapi-generator`. Currently there are other places where the naming is not purely Go idiomatic (see https://github.com/OpenAPITools/openapi-generator/issues/2740). 

Even though that were supported, I'm not a big fan of mixing the spec with a bunch of vendor extensions which are related to the Go client code generation, it seems like an anti-pattern. If we want to have better naming we should rather patch the tool generating the Go code to support more acronyms, uppercased properties, etc. In the meanwhile I think we should accept that naming isn't always 100% optimal.

**Comment by bamarni** [comment]
LGTM, calling the subclient `Metatada` looks great as it's in sync with https://docs.mesosphere.com/1.13/api/master-routes/#metadata

Could you maybe add a new metadata folder in the examples with a call to that endpoint? That's the only "integration tests" we have at the moment :sweat_smile:

**Pull request merged**